threading/smartptrs.nim

```diff
@@ -89,22 +89,20 @@ type
     ## Shared ownership reference counting pointer.
     val: ptr tuple[value: T, counter: Atomic[int]]
 
+proc decr[T](p: SharedPtr[T]) {.inline.} =
+  if p.val != nil:
+    # this `fetchSub` returns current val then subs
+    # so count == 1 means we're the last
+    if p.val.counter.fetchSub(1, Release) == 1:
```

Comment:
What's wrong with keep starting with 0 and test against `0` here?

Reply:
Nothing really. The only reason for the change was that I copied the setup from a solution I'd done elsewhere. Starting from 1 there helped me reason about the race conditions.

Reply:
```suggestion
    if p.val.counter.fetchSub(1, Release) == 0:
```